daemon.go

```diff
@@ -250,19 +250,18 @@ func peerAddrsInDHT(ctx context.Context, d kademlia, messenger *dhtpb.ProtocolMe
 	return addrMap, nil
 }
 
-func providerRecordInDHT(ctx context.Context, d kademlia, c cid.Cid, p peer.ID) bool {
+func providerRecordInDHT(ctx context.Context, d kademlia, c cid.Cid) bool {
```

Comment:
What's the idea behind the change here? This version seems to answer the question "has anyone advertised the record" which is different from "did node X that is hosting the data advertise it has the data".

Unless this is part of bigger changes towards #24 or #6, seems like we'd rather answer the second question. Otherwise, someone might see:
✔ Successfully connected to multiaddr
✔ Found multiaddr with 2 dht peers
✔ Found the multihash in the dht
✔ Successfully downloaded the CID from the peer

and yet the data would still not be retrievable because a different peer (who is unreachable) had advertised themselves in the DHT but this one had not

Reply:
Ah, okay. I will revert this.

The test tests if the first bootstrapper on the machine has an empty directory. Maybe there's been a change to our bootstrappers, which are no longer announcing CIDs? I have no idea. Wdyt?

My idea would be to just remove this test. We already check if the peer that we know "has" wikipedia returns `CidInDht` as true.

Reply:
@aschmahmann I'm also more than happy on porting some of the features and address #24 and #6.